services/dedup/dedup.go

```diff
@@ -17,9 +17,18 @@ import (
 	"github.com/rudderlabs/rudder-server/utils/misc"
 )
 
-type DedupI interface {
-	FindDuplicates(messageIDs []string, allMessageIDsSet map[string]struct{}) (duplicateIndexes []int)
-	MarkProcessed(messageIDs []string) error
+type Info struct {
+	MessageID string
+	Size      int
+}
+
+type Payload struct {
+	Size int `json:"size"`
+}
```

Comment:
We shouldn't need a struct for holding the payloadSize, an `int64` would suffice
```suggestion
```
For converting to-from bytes we may use `strconv`, e.g.
```go
value := strconv.FormatInt(payloadSize, 10)
payloadSize, _ = strconv.ParseInt(string(value), 10, 0)
```

Reply:
Since the payload stored in badgerDB should be backward compatible, I was storing this payload.

Reply:
Not sure if just storing a string would work here.

Reply:
what is stored in badgerDB now? Isn't it `nil`?

Reply:
Yeah, right now it is nil. But in the future, if we are going to change the payload format, our current structure would not work?

Reply:
If we change it in the future, we may use a fallback if parsing the value fails

Reply:
But wouldn't it better to set the correct expectations from the code instead of handling the error?

Reply:
We are dropping the previous values when badgerDB starts, thus we shouldn't worry too much on the evolution of the key's value format. At this stage I would prefer focusing on simplicity and performance (compute & storage).